src/engine/GroupBy.h

```diff
@@ -317,16 +341,20 @@ class GroupBy : public Operation {
     }
 
     // Get the values of the grouped column in ascending order.
-    [[nodiscard]] std::vector<Id> getSortedGroupColumn() const;
+    [[nodiscard]] T<std::vector<Id>> getSortedGroupColumns() const;
```

Comment:
Maybe this can even be an `IdTable`, which implicitly handles all your needs.

Reply:
But then again maybe this complicates your code, and at some point we want to get rid of the sorting anyway...